test/grpc-creds/sa.boulder/cert.pem

```diff
@@ -1,19 +1,20 @@
 -----BEGIN CERTIFICATE-----
-MIIDKTCCAhGgAwIBAgIIHaeuruCOx1AwDQYJKoZIhvcNAQELBQAwIDEeMBwGA1UE
-AxMVbWluaWNhIHJvb3QgY2EgM2I4YjJjMCAXDTE5MDEzMTIwMDg0MVoYDzIxMDkw
-MTMxMjAwODQxWjAVMRMwEQYDVQQDEwpzYS5ib3VsZGVyMIIBIjANBgkqhkiG9w0B
-AQEFAAOCAQ8AMIIBCgKCAQEAywJTI+hX1RvKcOuD+g6/dy09tRpoO5NzWQZ3Mu/2
-IDmYRFFq7MCPvB2qPWKLNm/5gBrviy0c2IFlpiXypC6YSye8oax0eOQjqLgSDIM/
-MR/rOFBq5m/h9Fb2SVNNcfCfYlh+UXlBKzRMVdl6K3/l5iuxSuAyZsw9TWmHhBOU
-smqYTfqnjtclRu9yPDZmoYbyO0K+7ZIifFewJkBPQhP+YeK1PwSGr4/7V7ee68Ax
-8bPX68MznLjaBhwGXSs/vzoKpJRXxcrRRGxHiXjknynQA8iOnLmeRPy5gs+rHNtf
-wBC84CQ4taebpuUQwbM02HV84BHtPYSclXhb9F3OhIlacwIDAQABo3AwbjAOBgNV
-HQ8BAf8EBAMCBaAwHQYDVR0lBBYwFAYIKwYBBQUHAwEGCCsGAQUFBwMCMAwGA1Ud
-EwEB/wQCMAAwLwYDVR0RBCgwJoIKc2EuYm91bGRlcoILc2ExLmJvdWxkZXKCC3Nh
-Mi5ib3VsZGVyMA0GCSqGSIb3DQEBCwUAA4IBAQBntsWlFhN5UHYK7ok7nRNaO5e/
-e4gAOajaELXhirRLRmWXifM7ZcwykmM+bT6/nt8Z6B4sy1T3EGpPqnyiBkaXpc8K
-OH3x9RmPXfIvgBjch/fccFE25zv0p+0uwlHlZk1+8MoMG2qJfAPqqTfNcE1iHejz
-GYdxsmydIPH8ux3yalxyOyqk6twbjbTDVZffAkKKHz3IvD6ERGYUzhr8nt02YvUw
-GnloyTmpnmMqX5JJBnt3/K8tMuZnSVQhTmoGRdWuAM6uD6iwr6oKerOEY5uiASh6
-ShNWLF0PPv9O/6XxThd2UJ79/hJn7gMqfYIeFD5Xx9ZD+0oRHqRjiF4+3ci1
+MIIDMzCCAhugAwIBAgIILdXJR+8c3vgwDQYJKoZIhvcNAQELBQAwIDEeMBwGA1UE
```

Comment:
Why is this change necessary for the SA, but not for any of our other services? For example, I don't see IP addresses in the current ra.boulder test cert.

Also, these files are generated by `//test/grpc-creds/generate.sh`; the next time someone runs that script in order to add creds for a new service, it'll overwrite this change.

Reply:
Had a slight detour into #6276 when I regenerated every certificate adding IP SANs and started seeing handshake failures.

> Also, these files are generated by //test/grpc-creds/generate.sh; the next time someone runs that script in order to add creds for a new service, it'll overwrite this change.

Thanks! I've shipped an update to generate.sh which should address this.

> Why is this change necessary for the SA, but not for any of our other services? For example, I don't see IP addresses in the current ra.boulder test cert.

From the RA's perspective, as a client, it's connecting to the SA at a given IP address and port. The SA presents a certificate to RA that's issued by the specific root that RA trusts for gRPC connections and it contains the IP name that RA used to initiate its connection to the SA. This authenticates the SA, so it sends the SA its certificate for authentication.

From SA's perspective, it receives a certificate from RA that's issued by the specific root that SA trusts for gRPC connections and it contains a DNS name `ra.boulder` that's included in the SA ClientNames list. So the SA grants access.